### ISSUE TITLE
WebJobs still run in deployment slots when slot is shut down

### ISSUE THREAD

**Issue by makhdumi** [projectkudu/kudu#1886]
After swapping from staging to production, we always shut down the staging slot, but the continuous webjobs (not sure about scheduled webjobs) continue to run. This is despite them showing as "stopped" in the WebJobs section of the Azure Portal.

The only way to tell that they're running is by viewing the job logs - which are often truncated. This cost us dozens of hours trying to figure out where "rogue" processing was coming from, because Azure reported everything as stopped.

This issue is somewhat related to https://github.com/projectkudu/kudu/issues/1365, but I don't think it's addressed. Yes, adding the `WEBJOBS_STOPPED` setting as a slot-stickied value "fixes" the problem, but at the very least I think this needs to be made clear in the documentation. It doesn't make sense that a job should still continue to run when the site is stopped.

**Comment by davidebbo** [comment]
The reason is that WebJobs run under the SCM site, which is not affected by the stop command. Though I agree it's not intuitive that WebJobs don't stop when you stop the site.

**Comment by davidebbo** [comment]
It turns out that there is a way to stop both, via an API call. The call is a bit complex, but does work. If you're interested, I can come up with what it would look like in PowerShell.

**Comment by Crypth** [comment]
WEBJOBS_STOPPED does not stop cron scheduled by settings.job on  "on demand" jobs. Our setup have 2 webjobs, one triggered by settings.job which adds the job to an Azure Queue and one that is triggered from that queue. The queue listening job is stopped on the staging slot, but the cron scheduled one isn't, so we get 2 jobs added to the queue which is a bit weird.

It's not a huge deal, but i would like to know if it's by design or not? 
(Should we wait for a fix or should we design around it)

**Comment by davidebbo** [comment]
@Crypth would be best to open separate issue for this. This issue tracks that when the site is stopped, WebJobs are unaffected (regardless of the WebJobs type). Your issue is specific to cron triggered WebJobs (and I would have expected them to stop, but there may be a bug).

**Comment by Crypth** [comment]
@davidebbo right you are. I'll verify that's what happening again and open a new issue if so.

**Comment by timvandenhof** [comment]
I'm expecting a behavior where the jobs would have the same running state as the website they are hosted in:
- Website is running, webdeploy, configuration deploy, webjobs should start after deployment
- Website is stopped, webdeploy, configuration deploy, webjobs should be stopped (and start when website is started)

We're facing an issue that our webjob is already started, directly after the webdeploy before the configuration is ready. To prevent that we tried to create an empty deployment slot (which auto starts), stop the slot, webdeploy the webapi and continuous webjobs, transform and apply the configuration (appsettings), bring the slot up back online, swap it with production, stop the old slot and stop the old webjob.

**Comment by makhdumi** [comment]
@timvandenhof you can use the `WEBJOBS_STOPPED` webapp setting on the empty slot you create. See [here](https://github.com/projectkudu/kudu/wiki/WebJobs#configuration-settings)for documentation.

i.e. create/set it to 1 before deploys, and once your deploy is done, set it to 0. If I recall correctly, it works "live" and will start/stop jobs immediately after being changed.

**Comment by davidebbo** [comment]
See also https://github.com/projectkudu/kudu/wiki/Full-stopping-a-Web-App for a way to stop both the main site and scm site (which runs WebJobs and deployment).

Closing since there is no code issue here.
